submitit/core/core.py

```diff
@@ -587,7 +587,8 @@ def batch(self) -> tp.Iterator[None]:
             delayed_batch = self._delayed_batch
             self._delayed_batch = None
         if not delayed_batch:
-            raise RuntimeError('No submission happened during "with executor.batch()" context.')
+            warnings.warn('WARNING: No submission happened during "with executor.batch()" context.')
```

Comment:
I don't think "WARNING is necessary since it *is* a warning
By default this is a UserWarning btw, it should probably be a RuntimeWarning instead?